src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect } from 'react';
+import { DataSourceSelector } from './datasource_selector';
+import { DataSourceType } from '../datasource_services';
+import { DataSourceGroup, DataSourceOption, DataSourceSelectableProps } from './types';
+import { ISourceDataSet } from '../datasource/types';
+import { IndexPattern } from '../../index_patterns';
+
+// Mapping between datasource type and its display name.
+const DATASOURCE_TYPE_DISPLAY_NAME_MAP = {
+  DEFAULT_INDEX_PATTERNS: 'Index patterns',
+  s3glue: 'Amazon S3',
+  spark: 'Spark',
+};
+
+type DataSetType = string | IndexPattern;
+
+export const DataSourceSelectable = ({
+  dataSources, // list of all available datasource connections.
+  dataSourceOptionList, // combo box renderable option list derived from dataSources
+  selectedSources, // current selected datasource in the form of [{ label: xxx, value: xxx }]
+  setSelectedSources,
+  setDataSourceOptionList,
+  onFetchDataSetError, //   onFetchDataSetError, Callback for handling dataset fetch errors. Ensure it's memoized.
+  singleSelection = true,
```

Comment:
I feel we could keep this as it is because it actually delegates how to handle multiple datasources selection to caller. Even we remove now and add later it will be the same at least from here. And now all usages with this are explicitly specifying single selection which should be fine.

Reply:
We should be careful in exposing features we do not support. If we dont support multiple datasource selection right now, we should not expose props for the same. Again, adding it is trivial and we will probably need additional changes that make more sense when we are ready for it. Right now this is a prop that assumes that the user will want to control the combobox like this. That assumption could be wrong but now its a part of the API and we have to support this prop as well.